api/go.mod

```diff
@@ -0,0 +1,74 @@
+module github.com/openshift/hypershift/api
+
+go 1.21
+
+require (
+	github.com/google/go-cmp v0.6.0
```

Comment:
Are you sure these are all the newest versions? I know the k8s ones have a newer version. Just trying to prevent more dependabot PRs :)